src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -524,75 +526,69 @@ export const handleStreamEncryptedResponses: RequestHandler<
  */
 export const handleGetEncryptedResponse: RequestHandler<
   { formId: string },
-  unknown,
+  EncryptedSubmissionDto | ErrorDto,
   unknown,
   { submissionId: string }
 > = async (req, res) => {
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
   const { submissionId } = req.query
   const { formId } = req.params
 
-  const logMeta = {
-    action: 'handleGetEncryptedResponse',
-    submissionId,
-    formId,
-  }
-
-  // Step 1: Retrieve submission.
-  const submissionResult = await getEncryptedSubmissionData(
-    formId,
-    submissionId,
-  )
-
-  if (submissionResult.isErr()) {
-    logger.error({
-      message: 'Failure retrieving encrypted submission from database',
-      meta: logMeta,
-      error: submissionResult.error,
-    })
-
-    const { statusCode, errorMessage } = mapRouteError(submissionResult.error)
-    return res.status(statusCode).json({
-      message: errorMessage,
-    })
-  }
+  return (
+    // Step 1: Retrieve logged in user.
+    getPopulatedUserById(sessionUserId)
+      // Step 2: Check whether user has read permissions to form.
+      .andThen((user) =>
+        getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Read,
+        }),
+      )
+      // Step 3: Check whether form is encrypt mode.
+      .andThen(checkFormIsEncryptMode)
+      // Step 4: Is encrypt mode form, retrieve submission data.
+      .andThen(() => getEncryptedSubmissionData(formId, submissionId))
+      // Step 5: Retrieve presigned URLs for attachments.
+      .andThen((submissionData) => {
+        // Remaining login duration in seconds.
+        const urlExpiry = (req.session?.cookie.maxAge ?? 0) / 1000
+        return transformAttachmentMetasToSignedUrls(
+          submissionData.attachmentMetadata,
+          urlExpiry,
+        ).map((presignedUrls) => {
+          // Successfully retrieved both submission and transforming presigned
+          // URLs, create and return new DTO.
+          const responseData: EncryptedSubmissionDto = {
+            refNo: submissionData._id,
+            submissionTime: moment(submissionData.created)
+              .tz('Asia/Singapore')
+              .format('ddd, D MMM YYYY, hh:mm:ss A'),
+            content: submissionData.encryptedContent,
+            verified: submissionData.verifiedContent,
+            attachmentMetadata: presignedUrls,
+          }
+          return responseData
+        })
```

Comment:
nit: maybe can just make this a utility method 
```ts

const generateEncryptedSubmissionDto = (submissionData, presignedUrls) => {blah}
```

main benefit would be readability